src/Menu.js

```diff
@@ -20,44 +18,52 @@ function generateSubmenuId() {
   return `sl-menu__submenu_${submenuIndex}`;
 }
 
+function isVisible(element) {
+  if (!element) return false;
+  const style = getComputedStyle(element);
+  return style.display !== 'none' && style.visibility !== 'hidden' && element.offsetParent !== null;
+}
+
 class MenuItem {
-  constructor(menu, $element) {
+  constructor(menu, element) {
     this.menu = menu;
-    this.$element = $element;
+    this.element = element;
     // Allow the link to be in a wrapper for styling purposes.
-    this.$link = $element.find('a').first();
+    this.link = element.querySelector(':scope > a');
     this.open = false;
 
-    if (!this.$element.attr('role')) {
-      this.$element.attr('role', 'none');
+    if (!this.element.getAttribute('role')) {
+      this.element.setAttribute('role', 'none');
     }
-    if (!this.$link.attr('role')) {
-      this.$link.attr('role', 'menuitem');
```

Comment:
Wait a minute... Is this for real? `jQuery` has a method that is a getter if you supply on arg, but a setter if you supply 2? 🤯 WHAT COULD GO WRONG?